calico/v3.28.0/apiserver/rockcraft.yaml

```diff
@@ -0,0 +1,52 @@
+name: calico-apiserver
+summary: Calico apiserver ROCK image.
+description: >
+  This rock is a drop in replacement for the
+  docker.io/calico/apiserver:v3.28.0 image.
+version: v3.28.0
+license: Apache-2.0
+
+base: ubuntu@24.04
+build-base: ubuntu@24.04
+platforms:
+  amd64:
+
+entrypoint-service: apiserver
+services:
+  apiserver:
+    command: /code/apiserver [ ]
+    override: replace
+    startup: enabled
+
+parts:
+  build-deps:
+    plugin: nil
+    build-snaps:
+      - go/1.22/stable
+
+  apiserver:
+    after: [build-deps]
+    plugin: go
+    source-type: git
+    source: https://github.com/projectcalico/calico
+    source-tag: v3.28.0
+    source-depth: 1
+    override-build: |
+      PACKAGE_NAME=github.com/projectcalico/calico/apiserver
```

Comment:
Can we move constant build environment variables to the `build-environment` part property?
Such as `PACKAGE_NAME`, `GIT_VERSION`, `CGO_ENABLED`